paimon-core/src/main/java/org/apache/paimon/metrics/commit/CommitMetrics.java

```diff
@@ -0,0 +1,280 @@
+                            () -> latestCommit.getPartitionedChangelogFilesAppended(partition));
+                    group.gauge(
+                            LAST_CHANGELOG_FILES_COMMIT_COMPACTED,
+                            () -> latestCommit.getPartitionedChangelogFilesCompacted(partition));
+                    group.gauge(
+                            LAST_DELTA_RECORDS_APPENDED,
+                            () -> latestCommit.getPartitionedDeltaRecordsAppended(partition));
+                    group.gauge(
+                            LAST_CHANGELOG_RECORDS_APPENDED,
+                            () -> latestCommit.getPartitionedChangelogRecordsAppended(partition));
+                    group.gauge(
+                            LAST_DELTA_RECORDS_COMMIT_COMPACTED,
+                            () -> latestCommit.getPartitionedDeltaRecordsCompacted(partition));
+                    group.gauge(
+                            LAST_CHANGELOG_RECORDS_COMMIT_COMPACTED,
+                            () -> latestCommit.getPartitionedChangelogRecordsCompacted(partition));
+                }
+            }
+        }
+    }
+
+    private class LatestDurationGauge implements Gauge<Long> {
+        @Override
+        public Long getValue() {
+            CommitStats commit = latestCommit;
+            if (commit != null) {
+                return commit.getDuration();
+            } else {
+                return 0L;
+            }
+        }
+    }
+
+    private class LatestAttemptsGauge implements Gauge<Long> {
+        @Override
+        public Long getValue() {
+            CommitStats commit = latestCommit;
+            if (commit != null) {
+                return commit.getAttemps();
+            } else {
+                return 0L;
+            }
+        }
+    }
+
+    private class LatestGeneratedSnapshotGauge implements Gauge<Long> {
+        @Override
+        public Long getValue() {
+            CommitStats commit = latestCommit;
+            if (commit != null) {
+                return commit.getGeneratedSnapshots();
+            } else {
+                return 0L;
+            }
+        }
+    }
+
+    private class LatestPartitionsWrittenGauge implements Gauge<Long> {
+        @Override
+        public Long getValue() {
+            CommitStats commit = latestCommit;
+            if (commit != null) {
+                return commit.getNumPartitionsWritten();
+            } else {
+                return 0L;
+            }
+        }
+    }
+
+    private class LatestBucketsWrittenGauge implements Gauge<Long> {
+        @Override
+        public Long getValue() {
+            CommitStats commit = latestCommit;
+            if (commit != null) {
+                return commit.getNumBucketsWritten();
+            } else {
+                return 0L;
+            }
+        }
+    }
+
+    public void reportCommit(CommitStats commitStats) {
+        commitMetricsReadWriteLock.lock();
```

Comment:
`public synchronized void`. No need for a `ReentrantLock`.

Also, `reportCommit` is only called in `FileStoreCommitImpl#commit`. This method should only be called in a single thread, so there is no need to lock this method.